packages/kit/src/runtime/client/client.js

```diff
@@ -1734,14 +1734,30 @@ export function create_client(app, target) {
 					});
 				});
 
+				/** @type {Array<import('./types').BranchNode | undefined>} */
+				const branch = await Promise.all(branch_promises);
+
+				const parsed_route = routes.find(({ id }) => id === route.id);
+
+				// server-side will have compacted the branch, reinstate empty slots
+				// so that error boundaries can be lined up correctly
+				if (parsed_route) {
```

Comment:
Instead of having this additional code to un-compact it, wouldn't it be less code to not compact the branch server-side for the initial request?

Reply:
Need to litter quite a few null checks to do it from the server side, https://github.com/sveltejs/kit/commit/cb3dcdeca69268042814997ac50089a6a5c2047a

I was thinking it'd better to have slightly more code in a perma-cached asset than adding empty slots for the node_ids and the corresponding server data nodes in every SSR'ed response.